### DISCUSSION TITLE
Why need download other platform packages?

### DISCUSSION THREAD

**Discussion by tjx666** [swc-project/swc#5872] [Q&A] [answered]
My computer is MacOS, why need to install other platform packages?

![image](https://user-images.githubusercontent.com/41773861/190326178-a97bf595-8860-4423-8598-f8c252bd4c48.png)

**Comment by kwonoj** [comment, marked as answer]
This question should go for the upstream package manager. SWC specifies per-platform optional dependencies which expect to work with npm and some others, if some pkg manager does not honor SWC itself cannot control how.

**Comment by tjx666** [comment]
@zkochan any idea about this?

**Reply by zkochan** [reply]
It happens because you use the `--force` option.

**Reply by tjx666** [reply]
I just want to overwrite local packages because I modify the file under node_modules. I think --force should not install other platform, because installing other platform packages seems useless in any situation. Or there any better chooses for me to overwrite local packages?

**Reply by Brooooooklyn** [reply]
> because installing other platform packages seems useless in any situation

It's useful for cross-compiling, like build docker images/electron app for another platform
